src/dune_rules/pkg_rules.ml

```diff
@@ -266,7 +270,10 @@ module Pkg = struct
 
   let top_closure depends =
     match
-      Top_closure.top_closure depends ~key:(fun t -> t.id) ~deps:(fun t -> t.depends)
+      Top_closure.top_closure
+        depends
+        ~key:(fun t -> t.id)
+        ~deps:(fun t -> t.depends @ t.inherited_post_depends)
```

Comment:
This looks wrong to me. I would imagine this to look like:

```ocaml
~deps:(fun t -> t.depends @ List.concat_map t.depends ~f:(fun d -> d.post))
```

Reply:
How is that different than what I've written? The `t.inherrited_post_depends` field contains the union of post dependencies of each dependency of `t` (apart from `t` itself).

Reply:
Oh and I should clarify that the reason I added a field with the union of post dependencies of all deps rather than just adding a field with the package's own post deps is it simplifies handling dependency cycles broken by post dependencies.

Reply:
> How is that different than what I've written? The t.inherrited_post_depends field contains the union of post dependencies of each dependency of t (apart from t itself).

I'd like to think it's the same, but I'm not sure - that's the problem. Doing it two stages converted a change that is simple to specify into something I need to think about. Let me get back to you on that.